docs/current/cookbook.md

```diff
@@ -34,31 +34,6 @@ The following code listing obtains a reference to the host working directory and
 </TabItem>
 </Tabs>
 
-When the Dagger pipeline code is in a sub-directory, it may be more useful to set the parent directory (the project's root directory) as the working directory.
```

Comment:
I'm not sure about removing this one, however I couldn't find a way to make it run the same way with and without `dagger run`. In case one of the reviewers has a solution, please comment.

Reply:
there's something we can do to detect if the user is running the pipeline via `dagger run`. That'd be checking if the `DAGGER_SESSION_TOKEN` exists. 

```go 

func main() {
	ctx := context.Background()

	args := []dagger.ClientOpt{}
	if _, ok := os.LookupEnv("DAGGER_SESSION_TOKEN"); !ok {
		// running pipeline via `dagger run`
		args = append(args, dagger.WithWorkdir("/tmp"))
	}

	client, err := dagger.Connect(ctx, args...)
	if err != nil {
		panic(err)
	}

	_, err = client.Container().From("alpine").WithMountedSecret("/secret/foo", client.Secret("foosecret")).
		WithExec([]string{"cat", "/secret/foo"}).
		Sync(ctx)

	fmt.Println(err)
}
```

WDYT?

Reply:
It's the actual purpose of this example to show how use workdir :/
That sound complex to work around, but you can add a warning (just for this one because it's special)

Reply:
What if we modify the sdk to just warning if we run the client with dagger run and set workdir option? Or do somehow to passing workdir to the sdk (like a env var) and warning it if it’s not the same. 🤔

Reply:
We are actually throwing an error if you try running `dagger run --workdir <path>` with a workdir also set on the SDK side

Reply:
> there's something we can do to detect if the user is running the pipeline via `dagger run`. That'd be checking if the `DAGGER_SESSION_TOKEN` exists.

I read and found this interesting. I don't think we need it in this example but it could perhaps be a different one. However the use case is not clear to me ie. why would a user need to perform this detection, if the ideal path is to always use `dagger run`?